### ISSUE TITLE
RUCSS - PHP warning: Invalid argument supplied for foreach() > wp-background-process.php > line 314

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4230]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ `3.9.1.1`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

A customer reported the following error related to the Remove Unused CSS feature:
```
Got error 'rocessing/wp-background-process.php on line 314
PHP message:
PHP Warning: Invalid argument supplied for foreach() in /var/www/virtual/example.com/frontend/wp-content/plugins/wp-rocket/inc/classes/dependencies/wp-media/background-processing/wp-background-process.php on line 314
```
This is happening here:
https://github.com/wp-media/wp-rocket/blob/7bc0dd8dcc0ef5e2af156d978c8a437ec8bb1bd1/inc/classes/dependencies/wp-media/background-processing/wp-background-process.php#L314

and it's happening when the `ResourceFetcherProcess` class is used: ` /wp-admin/admin-ajax.php?action=rocket_rucss_warmup_resource_fetcher`

They are using:
* a load balancer
* Redis
* Varnish

**To Reproduce**
Although they were quite a lot of such entries in their logs, the customer had trouble reproducing the error manually.

**Expected behavior**
No PHP warnings should pop up when using Remove Unused CSS.

**Additional context**

**Ticket:** https://secure.helpscout.net/conversation/1590090544/284922?folderId=2135277
**Slack conversation:** https://wp-media.slack.com/archives/C43T1AYMQ/p1628078823007400

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by jespervnielsen** [comment]
Hello.

I am the "customer"

I can see in the ticket description, that you say its related to the "remove unused CSS" - but that is'nt activated.
![image](https://user-images.githubusercontent.com/31649178/128668370-e42657a9-430c-4c29-a720-e1dd8a78f716.png)

The reason i can't trigger the error manually, is because i dont know what triggers the "ResourceFetcherProcess".
How is that usually called? - which functionality?
How can i trigger it? can i trigger it, via a ajax-requrest, via saving a post, via CLI, ... other?

**Comment by vmanthos** [comment]
@jespervnielsen 

The `ResourceFetcherProcess` is used by the Remove Unused CSS feature.

It's called here:
https://github.com/wp-media/wp-rocket/blob/7bc0dd8dcc0ef5e2af156d978c8a437ec8bb1bd1/inc/Engine/Optimization/RUCSS/Warmup/Subscriber.php#L94
when the feature is active. 

To trigger it you should enable Remove Unused CSS.

> I can see in the ticket description, that you say its related to the "remove unused CSS" - but that is'nt activated.

In the ticket you opened with us, you mentioned the warning was related to `ResourceFetcherProcess`. This means the feature was active.

Are you keep getting these notices even when the feature is deactivated? Is this deactivated in all instances of the load balancer?

**Comment by jespervnielsen** [comment]
@vmanthos 

The screenshot is from this morning.
And in the last 2 hours, there has been 10,046 messages
![image](https://user-images.githubusercontent.com/31649178/128672717-f4eb02a0-1d49-43f7-84ff-16d0bc627ff1.png)

And according to "stream" (backend user activity logging tool https://wordpress.org/plugins/stream/ ) the latest update of the WP Rocket settings was 16 days ago..

The reason why I suspected the "ResourceFetcherProcess" is because, if I take a log entry with the error, and looks at the request ID. (all logs, generated by a http request, has a relation to the request) - then i can see its triggered by the below POST request. - and that mentions the "rocket_rucss_warmup_resource_fetcher" (And that request is generated, after I took the screenshot of the setting)

POST wp-admin/admin-ajax.php?action=rocket_rucss_warmup_resource_fetcher&nonce=400841f76c HTTP/1.1

So for me, its sounds wierd, if the ajax request is being called, if the feature is disabled? - unless its two different features?

And yes - the feature is deactivated on all instances of the loadbalancer. - The DB is the same across the instances. it has its own server. - so all webinstances connect to the same DB.
The loadbalancer just splits the load of the request across the webservers.

**Comment by spk100** [comment]
I have the same issue with the new version.

Wordfence has been going crazy. I have more than 1000 alerts from Wordfence 

Activity Detail
Tampa, Florida, United States left https://www.mysite.com/wp-admin/admin-ajax.php?action=rocket_rucss_warmup_resource_fetcher&n… and was blocked by firewall for LFI: Local File Inclusion in POST body: 0=%2Fhome%2Fmysite%2Fpublic_html%2Fwp-content%2Fcache%2Fmin%2F3rd-party%2Fpagead2.googlesyndication.… at https://www.mysite.com/wp-admin/admin-ajax.php?action=rocket_rucss_warmup_resource_fetcher&nonce=…
8/26/2021 9:19:46 PM (1 minute ago) 


Been getting alerts like this non-stop.

Site name removed for obvious reasons.

**Comment by spk100** [comment]
The latest version still has the same issue 
The Wordfence Web Application Firewall has blocked 176 attacks over the last 10 minutes. Below is a sample of these recent attacks:September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:33pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:32pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:32pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:32pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...
September 16, 2021 1:32pm  23.111.145.18 (United States)     Blocked for LFI: Local File Inclusion in POST body: 0 = /home/xyz/public_html/wp-content/cache/min/3rd-party/pagead2.googlesyndication.com-pagead-js-ad...

**Comment by piotrbak** [comment]
It's fixed in 3.11:
1. We're not using the WP Background Processing anymore
2. We're not warming up the resources
